### DISCUSSION TITLE
Easiest way to get page id?

### DISCUSSION THREAD

**Discussion by peteruithoven** [requarks/wiki#5471] [Help / Questions] [answered]
What's the easiest way to get the id of a page? 

I know I can use the page source (`/s/`), so to get the page id of `/en/ops/people` I go to: `/s/en/ops/people`, then in the top right the page id is visible. But this is quite a hassle. 

I want to know the page id so I could make links based on page id's, so that changing the url doesn't break links. This means instead of linking to `/en/ops/people` I'd use `/i/296`. 

Relevant:
- https://js.wiki/feedback/p/persistent-links
- https://js.wiki/feedback/p/moving-a-page-should-update-links
- https://js.wiki/feedback/p/setting-to-enable-id-based-links

**Comment by smcintosh80** [comment]
This GraphQL query will get what you want:

```
{
  pages {
    list {
      id
      path
      title
    }
  }
}
```

**Reply by peteruithoven** [reply]
Thanks, but I can't ask my less technical colleagues to run graphql queries ;)

**Comment by NGPixel** [comment, marked as answer]
There's no easy way to get the ID at the moment.
